gammapy/makers/background/reflected.py

```diff
@@ -45,6 +45,40 @@ def are_regions_overlapping_rad_max(regions, rad_max, offset, e_min, e_max):
     return np.any(separations[np.newaxis, :] < (2 * rad_max_at_offset))
 
 
+def is_rad_max_compatible_region_geom(rad_max, geom, rtol=1e-3):
+    """Check if input RegionGeom is compatible with rad_max for point-like analysis.
+
+    Parameters
+    ----------
+    geom : `~gammapy.maps.RegionGeom`
+        input RegionGeom.
+    rtol : float
+        relative tolerance
+
+    Returns
+    -------
+    valid : bool
+        True if rad_max is fixed and region is a CircleSkyRegion with compatible radius
+        True if region is a PointSkyRegion
+        False otherwise.
+    """
+    if geom.is_all_point_sky_regions:
+        valid = True
+    elif isinstance(geom.region, CircleSkyRegion) and rad_max.is_fixed_rad_max:
+        valid = np.allclose(geom.region.radius, rad_max.quantity, rtol)
+
+        if not valid:
+            raise ValueError(
```

Comment:
Here the logic is problematic too. The behavior is inconsistent between the test raising an exception for a specific set of parameters and returning a boolean flag the rest of the time.